flow/aggregate.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+from collections.abc import Iterable
+from copy import deepcopy
+from itertools import groupby
+from itertools import zip_longest
+from functools import partial
+
+import signac
+from signac.contrib.project import Project, JobsCursor
+
+
+class aggregate:
+    """Decorator for operation functions that are to be aggregated.
+
+    If this class is used for aggregation then by-default, if the aggregator
+    parameter is not passed, an aggregate of all jobs will be created.
+
+    .. code-block:: python
+
+        example_aggregate = aggregate()
+        @example_aggregate
+        @FlowProject.operation
+        def foo(*jobs):
+            print(len(jobs))
+
+    :param aggregator:
+        Information on how to aggregate jobs. Takes in a list of
+        jobs and can return or yield lists or single jobs instead.
+        The default behavior is creating an aggregate of all jobs.
```

Comment:
```suggestion
        Information describing how to aggregate jobs. Takes in a list of
        jobs and can return or yield lists or single jobs instead.
        The default behavior is creating an aggregate of all jobs.
```

Reply:
```suggestion
        Information describing how to aggregate jobs. Is a callable that
        takes in a list of jobs and can return or yield subsets of jobs as list
         or single jobs. The default behavior is creating a single aggregate
         of all jobs.
```
Also, I would recommend that the function should always return list or tuples of subsets of jobs. The yielded or returned value should be an iterable.